crates/sui-indexer-alt/src/handlers/tx_kinds.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use anyhow::Result;
+use diesel_async::RunQueryDsl;
+use sui_types::full_checkpoint_content::CheckpointData;
+
+use crate::{db, models::transactions::StoredTxKind, schema::tx_kinds};
+
+use super::Handler;
+
+pub struct TxKinds;
+
+#[derive(Debug, Clone)]
+pub enum TxKind {
+    SystemTransaction = 0,
+    ProgrammableTransaction = 1,
+}
```

Comment:
nit: This should be under `models::transactions`.

Reply:
Also, I wonder whether you could just store the `TxKind` directly, if you give it the correct `repr`? (I have no idea, genuinely curious whether diesel would pick that up).

```suggestion
#[derive(Debug, Clone)]
#[repr(i16)]
pub enum TxKind {
    SystemTransaction = 0,
    ProgrammableTransaction = 1,
}
```

Reply:
`repr` does not seem to do much -- it only tells Rust how to store this enum in memory but casting is still necessary.

Reply:
I know that we need to cast it if it's going into an `i16` type, but I was wondering whether `diesel` would do something clever if in the model type, you used `TxKind` directly, and set its `repr` to `i16`.